modules/Sources/Features/Root/RootInitialization.swift

```diff
@@ -38,14 +39,53 @@ extension RootReducer {
             switch action {
             case .initialization(.appDelegate(.didEnterBackground)):
                 sdkSynchronizer.stop()
-                return .none
+                state.bgTask?.setTaskCompleted(success: false)
+                state.bgTask = nil
```

Comment:
[not-blocking] _Note: the docs for background processing are confusing so this could be a non-issue._

I can't find good documentation on Apple docs about the exact system events sequence of initializing an app into the background directly.

[This pieces of documentation](https://developer.apple.com/documentation/uikit/app_and_environment/scenes/preparing_your_ui_to_run_in_the_background/about_the_background_execution_sequence) makes me think these lines of code could be completing background tasks beforehand.

[Also this other doc implies so](https://developer.apple.com/documentation/uikit/app_and_environment/scenes/preparing_your_ui_to_run_in_the_background/about_the_background_execution_sequence)

```
If your app is in memory and suspended when an event arrives, the system resumes the app in the background, following this sequence:

The system resumes the app.

UIKit calls the app delegate’s [applicationDidEnterBackground(_:)](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622997-applicationdidenterbackground) method.

UIKit delivers the event that caused the launch.

The app’s snapshot is taken.

The app may be suspended again.
```

Reply:
Confused me as well at the beginning but never happened to me during the testing. The sequence is exactly how it's expected. One time I observed the backgroundNotification to come but closure for the background task was initiated afterwards, resulting in the right order or operations.